### ISSUE TITLE
BLD: move asv build to a daily travis cron

### ISSUE THREAD

**Issue by jreback** [pandas-dev/pandas#19695]

**Comment by TomAugspurger** [comment]
Are you working on this, or should I?

**Comment by jreback** [comment]
feel free!

I would create another repo (in pandas-dev) to do this 
similar to the wheel building 

i suppose u could reuse that code

**Comment by TomAugspurger** [comment]
Good idea. Thoughts on moving over

1. ASV
2. conda-build
3. pip build

Anything else?

**Comment by jreback** [comment]
i think those plus 32 bit build would be awesome

maybe just run these on a daily cron

**Comment by jreback** [comment]
maybe doc build as well?

**Comment by TomAugspurger** [comment]
I think Joris had plans (or hopes?) to move the doc build to circle-CI and
upload an artifact so that users can check the doc build, similar to how
scikit-learn does it.

**Comment by jreback** [comment]
even better!

note that we do this also in ibis-project

**Comment by TomAugspurger** [comment]
👍 . I'll be offline for a couple hours, but will start on this afterwards.

Tangentially related, I'd like to go through all our `ci/*` stuff and simplify / consolidate our dependency code.

I want to get to the point where we have a single `environment.yaml` per build, and *try* to get everything installed in a single shot by `conda`, instead of multiple steps. That'll help with previously pinned deps getting upgraded without us knowing.

And then we can small script that will print out the various versions of deps we're tested against.

I'd also like to better manage optional dependencies and test-only dependencies, but not sure the best way forward yet.

**Comment by jreback** [comment]
all for cleaning up
but be careful
a lot of that code is designed to give informative errors when things fail and to no pin things too much

**Comment by jorisvandenbossche** [comment]
> I think Joris had plans (or hopes?) to move the doc build to circle-CI and upload an artifact so that users can check the doc build, similar to how scikit-learn does it.

Even apart from that, I often check the doc build to then comment on the PR with things that break the docs, so I would prefer to keep this in the pandas CI itself.

**Comment by TomAugspurger** [comment]
FYI: https://travis-ci.org/pandas-dev/pandas-ci/ if you want to sign up for
alerts on build failures, though maybe wait until I get a successful build
first :) Then I'll switch it over to a daily cron.

I'm doing conda build and ASV now.

Pip build with no Cython will follow once we figure out why it's broken.

**Comment by jreback** [comment]
yeah not concerned about the doc build really. the other builds are just taking extra time on each commit and doing daily would be sufficient.
